src/Umbraco.Cms.Integrations.Crm.Hubspot/Client/src/property-editor/authorization-property-editor.element.ts

```diff
@@ -0,0 +1,189 @@
+﻿import { UmbElementMixin } from "@umbraco-cms/backoffice/element-api";
+import {
+    LitElement,
+    customElement,
+    html,
+    property,
+    state,
+    when
+} from "@umbraco-cms/backoffice/external/lit";
+import {
+    UMB_NOTIFICATION_CONTEXT,
+    type UmbNotificationColor,
+} from "@umbraco-cms/backoffice/notification";
+import {
+    ConfigDescription,
+    type HubspotOAuthSetup,
+    type HubspotServiceStatus
+} from "../models/hubspot-service.model.js";
+import { HUBSPOT_FORMS_CONTEXT_TOKEN } from "@umbraco-integrations/hubspot-forms/context";
+import type { OAuthRequestDtoModel } from "@umbraco-integrations/hubspot-forms/generated";
+
+const elementName = "hubspot-authorization";
+
+@customElement(elementName)
+export class HubspotAuthorizationElement extends UmbElementMixin(LitElement) {
+
+    #hubspotFormsContext!: typeof HUBSPOT_FORMS_CONTEXT_TOKEN.TYPE;
+
+    @state()
+    private _serviceStatus: HubspotServiceStatus = {
+        isValid: false,
+        type: "",
+        description: "",
+        useOAuth: false
+    };
+
+    @state()
+    private _oauthSetup: HubspotOAuthSetup = {
+        isConnected: false,
+        isAccessTokenExpired: false,
+        isAccessTokenValid: false
+    };
+
+    @property({ type: String })
+    public value = "";
+
+    constructor() {
+        super();
+
+        this.consumeContext(HUBSPOT_FORMS_CONTEXT_TOKEN, (context) => {
+            this.#hubspotFormsContext = context;
+        });
+    }
+
+    async connectedCallback() {
+        super.connectedCallback();
+        await this.#checkApiConfiguration();
+    }
+
+    async #checkApiConfiguration() {
+        const { data } = await this.#hubspotFormsContext.checkApiConfiguration();
```

Comment:
Could consider moving this back to the context, and making the serviceStatus object an observable. Then the dependent components can observe that object rather than having to request the API configuration

Reply:
Moved the object into the context and `observe` it from the other components.